src/main/java/com/slack/kaldb/server/ArmeriaService.java

```diff
@@ -0,0 +1,107 @@
+package com.slack.kaldb.server;
+
+import com.google.common.util.concurrent.AbstractIdleService;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.grpc.GrpcMeterIdPrefixFunction;
+import com.linecorp.armeria.common.logging.LogLevel;
+import com.linecorp.armeria.server.Server;
+import com.linecorp.armeria.server.ServerBuilder;
+import com.linecorp.armeria.server.docs.DocService;
+import com.linecorp.armeria.server.grpc.GrpcService;
+import com.linecorp.armeria.server.grpc.GrpcServiceBuilder;
+import com.linecorp.armeria.server.healthcheck.HealthCheckService;
+import com.linecorp.armeria.server.logging.LoggingService;
+import com.linecorp.armeria.server.logging.LoggingServiceBuilder;
+import com.linecorp.armeria.server.management.ManagementService;
+import com.linecorp.armeria.server.metric.MetricCollectingService;
+import com.slack.kaldb.elasticsearchApi.ElasticsearchApiService;
+import com.slack.kaldb.proto.service.KaldbServiceGrpc;
+import io.micrometer.prometheus.PrometheusMeterRegistry;
+import io.netty.handler.codec.http.DefaultHttpHeaders;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.TimeUnit;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class ArmeriaService extends AbstractIdleService {
+  private final Logger LOG = LoggerFactory.getLogger(ArmeriaService.class);
+
+  private final PrometheusMeterRegistry prometheusMeterRegistry;
+  private final KaldbServiceGrpc.KaldbServiceImplBase searcher;
+  private final String serviceName;
+  private final Server server;
+  private final int serverPort;
+
+  public ArmeriaService(
+      int serverPort,
+      PrometheusMeterRegistry prometheusMeterRegistry,
+      KaldbServiceGrpc.KaldbServiceImplBase searcher) {
+    this(serverPort, prometheusMeterRegistry, searcher, null);
```

Comment:
instead of null can we pass `armeria` or something?

Reply:
This constructor isn't used currently, and when the name is null, it defaults to the class name which would be `ArmeriaService`.

I allowed changing the name here, so the logs are clearer if it's indexer armeria service or query armeria service (if both are running concurrently).

Reply:
got it! thanks for explaining